app/helpers/submissions_helper.rb

```diff
@@ -38,6 +38,13 @@ def set_release_on_results(groupings, release)
       Result.where(id: groupings.joins(:current_result).pluck('results.id'))
             .update_all(released_to_students: release)
     end
+    return if(not release)
+      groupings.each do |grouping|
+        grouping.accepted_students.each do |student|
+          NotificationMailer.with(user: student, grouping: grouping).release_email.deliver_now
+        end
+      end
+    end
```

Comment:
And you should remove this `end` as well. Basically, `return if ___` or `return unless ___` isn't a block, but a single statement. You can read more about these statements at https://www.natashatherobot.com/ruby-shorthands-if-then-else/.